clientv3/watch.go

```diff
@@ -643,12 +646,7 @@ func (w *watchGrpcStream) serveSubstream(ws *watcherStream, resumec chan struct{
 	// lazily send cancel message if events on missing id
 }
 
-func (w *watchGrpcStream) newWatchClient() (pb.Watch_WatchClient, error) {
-	// connect to grpc stream
-	wc, err := w.openWatchClient()
-	if err != nil {
-		return nil, v3rpc.Error(err)
-	}
+func (w *watchGrpcStream) newWatchClient() (ws pb.Watch_WatchClient, err error) {
```

Comment:
I think a new stream should go through the same path as a resume; there's no significant difference between creating a new stream and resuming a stream (aside from the resume having its initial revision updated). We used to have that distinction but it made reasoning about the resume path complicated and error-prone.

The function can be trimmed down by splitting out the goroutine phase in `newWatchClient` that wait for the cancels into a separate function (e.g., `waitOnResumeCancels`) and have `newWatchClient` call that.